### PR TITLE
Feat/liquidity tab charts 1101

### LINKED ISSUE #1101: Liquidity tab charts
[Specs](https://centrifuge.hackmd.io/JffFiSG2TnKHRQhOkHzJMw)

## Background
As an investor and issuer I want to see a chart where I can keep an overview of the incoming investments and redemptions by the investor as well as repayments and originations of loans by the issuer. By showing both charts the issuer and investor have full transparency on each other's actions. 

## Repayments & originations  (**pool liquidity tab**]
- **Origination** = When issuer borrows money from the pool (borrowing)
- **Repayments** = When issuer (partially and fully) repays a loan
- Positive/Negative bar chart
    - Repayments above zero
    - Originations below zero
- X-axis: 
    - Shows date of execution of the last 10 epochs 
- Y-Axis:
    -  Amount in pool currency (USDT/USDC) 

## Investments & redemptions **(pool liquidity tab)**
- **Investments** = when investors deposit money into the pool -> executed not locked
- **Redemptions** = When investors redeem their tokens from the pool -> executed not locked 
- X-axis: 
    - Shows date of execution of the last 10 epochs 
- Y-Axis: 
    - Amount in pool currency (USDT/USDC) 

## Location 
- add charts below reserve vs. cash drag chart

### PR DESCRIPTION
**Pull request by Hornebom**
### Description
Pool transactions charts for Pool liquidity tab. Custom stacked bar chart and csv download option.

Closes #1101

### Approvals
- [ ] Dev

### CODE REVIEW

File: centrifuge-app/src/utils/date.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-export function formatDate(timestamp: number | string | Date) {
+export function formatDate(timestamp: number | string | Date, options?: Intl.DateTimeFormatOptions) {
   return new Date(timestamp).toLocaleDateString('en-US', {
     year: 'numeric',
     month: 'short',
     day: 'numeric',
+    ...(options && options),
```
Comment (onnovisser) [inline centrifuge-app/src/utils/date.ts:6]: just `...options,` is enough

File: centrifuge-app/src/components/Charts/StackedBarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    bottom: number
+    date: string
+  }[]
+  names: [string, string]
+  colors: [string, string]
+  xAxisLabel?: string
+  currency: string
+}
+
+export function StackedBarChart({ data, names, colors, xAxisLabel, currency }: StackedBarChartProps) {
```
Comment (onnovisser) [inline centrifuge-app/src/components/Charts/StackedBarChart.tsx:34]: Nice, I like the abstraction

File: centrifuge-app/src/utils/date.ts (hunk 1 above)
Comment (Hornebom) [reply centrifuge-app/src/utils/date.ts]: Updated

File: centrifuge-app/src/utils/usePools.ts [hunk 3]
```diff
@@ -1,10 +1,18 @@
-import Centrifuge, { PoolMetadata } from '@centrifuge/centrifuge-js'
+import Centrifuge, { Pool, PoolMetadata } from '@centrifuge/centrifuge-js'
 import { useQuery } from 'react-query'
 import { combineLatest, map, Observable } from 'rxjs'
 import { useCentrifuge } from '../components/CentrifugeProvider'
 import { useCentrifugeQuery } from './useCentrifugeQuery'
 import { useMetadata } from './useMetadata'
 
+export function usePoolLiquidityTransactions(pool: Pool, fromEpoch: number, toEpoch: number) {
+  const [result] = useCentrifugeQuery(['poolsLiquidityTransactions', pool.id], (cent) =>
```
Comment (onnovisser) [inline centrifuge-app/src/utils/usePools.ts:9]: fromEpoch and toEpoch should also be in the keys array

Comment (Hornebom) [reply centrifuge-app/src/utils/usePools.ts]: 👌  updated

File: centrifuge-app/src/components/Charts/StackedBarChart.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import { Text } from '@centrifuge/fabric'
+import React from 'react'
```
Comment (onnovisser) [inline centrifuge-app/src/components/Charts/StackedBarChart.tsx:2]: `import * as React from 'react'

File: centrifuge-app/src/components/LiquidityTransactionsSection.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
+import { Pool } from '@centrifuge/centrifuge-js'
+import { AnchorButton, IconDownload, Text } from '@centrifuge/fabric'
+import React from 'react'
```
Comment (onnovisser) [inline centrifuge-app/src/components/LiquidityTransactionsSection.tsx:3]: same

File: centrifuge-app/src/components/Charts/StackedBarChart.tsx (hunk 4 above)
Comment (Hornebom) [reply centrifuge-app/src/components/Charts/StackedBarChart.tsx]: refactored

**Review by onnovisser — APPROVED** [review]

**Pull request merged**